mods/NERO/agent.py

```diff
@@ -1,202 +1,74 @@
+    group = 'Agent'
+    def __init__(self, team_type, *args):
+        self.team_type = team_type
 
     def initialize(self, init_info):
         self.actions = init_info.actions
         self.sensors = init_info.sensors
-        return True
-
-    def end(self, time, reward):
-        org = self.get_org()
-        # if a trace has been loaded, then process it here
-        if self.environment.use_trace:
-            trfitness = self.evaluate_trace()  # Calculate trace fitness and passed steps
-            if self.environment.run_backprop:  # backprop invalidates org.fitness because it changes network
-                if self.index != 0:  # run backprop only if the agent is not the elite organism
-                    self.backprop_trace()
-                    trfitness = self.evaluate_trace()  # Calculate trace fitness again since backprop changes network
-                org.fitness = trfitness
-                print "index: %d, trace fitness: %f" % (self.index, trfitness)
-            else:
-                org.fitness = floor(org.fitness if org.fitness > 0.0 else 0.0) + trfitness
-                print "index: %d, trace fitness: %f, combined fitness: %f" % (self.index, trfitness, org.fitness)
-        else:
-            print "index: %d, fitness: %f" % (self.index, org.fitness)
-        return True
-
-    def destroy(self):
-        env = OpenNero.get_environment()
-        if env is not None:
-            env.remove_agent(self)
-        return True
-
-        
-class KeyboardAgent(NeroAgent, OpenNero.AgentBrain):
-    keys = set([]) # keys pressed
-    action_map = {'fwd':0, 'back':1, 'left':2, 'right':3}
-    ACCELERATION = 0.1 # max acceleration of agent in one step
-    DRAG_CONST = 0.005 # constant for calculating drag
-    
-    ai = None
-
-    """
-    Keyboard agent with simple ASWD(QE) controls
-    """
-    def __init__(self):
-        """
-        this is the constructor - it gets called when the brain object is first created
-        """
-        # call the parent constructor
-        OpenNero.AgentBrain.__init__(self) # do not remove!
-        NeroAgent.__init__(self, group='Turret')
-
-    def initialize(self, init_info):
-        """
-        called when the agent is first created
-        """
-        self.actions = init_info.actions
-        return True
-
-    def start(self, time, sensors):
-        """
-        take in the initial sensors and return the first action
-        """
-        self.velocity = 0
-        return self.key_action()
-
-    def act(self, time, sensors, reward):
-        """
-        take in new sensors and reward from previous action and return the next action
-        """
-        return self.key_action()
-
-    def end(self, time, reward):
-        """
-        take in last reward
-        """
-        # store last reward and add average reward to fitness (i.e. how good the agent was since
-        # collecting the last cube)
+        self.rewards = init_info.sensors
```

Comment:
it seems weird to me that you are assigning init_info.sensors to self.rewards... what does that mean?